### ISSUE TITLE
[2.x][3.x] Remove BC layers when possible

### ISSUE THREAD

**Issue by alexislefebvre** [liip/LiipFunctionalTestBundle#520]
> Don't miss reverting depending on how we manage to fix https://github.com/symfony/symfony/issues/31762

_Originally posted by @nicolas-grekas in https://github.com/liip/LiipFunctionalTestBundle/pull/519_

Roadmap:

- [x] wait for next release of Symfony 4.3
- [x] set this next release of Symfony 4.3 as the minimal version for 4.3, or declare conflict with 4.3.0?
- [x] (2.x, less urgent) Revert #519 Feature/cherry pick 4.3 → see #527
- [x] (3.x) Revert #512 Add compatibility for Symfony 4.3+ → see #526

**Comment by gnutix** [comment]
@alexislefebvre symfony 4.3.1 has been released!

**Comment by alexislefebvre** [comment]
Now BC layers are broken by fixes that reverted BC breaks :upside_down_face:, see #519 and #521

**Comment by alexislefebvre** [comment]
Fixed in #522 and #523 but tests are broken. :thinking:

**Comment by alexislefebvre** [comment]
See #523 525

**Comment by alexislefebvre** [comment]
See #525

**Comment by alexislefebvre** [comment]
A new error appeared with Symfony 4.3.1 and is visible in #523:

> TypeError: Argument 1 passed to Symfony\Bundle\FrameworkBundle\Test\WebTestCase::getClient() must be an instance of Symfony\Bundle\FrameworkBundle\KernelBrowser or null, instance of Liip\FunctionalTestBundle\QueryCountClient given, called in /home/travis/build/liip/LiipFunctionalTestBundle/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Test/WebTestCase.php on line 55

(source: https://travis-ci.org/liip/LiipFunctionalTestBundle/jobs/542656030 )

It is triggered by the following configuration:

```yml
liip_functional_test:
    query:
        max_query_count: 1
```

**Comment by alexislefebvre** [comment]
Please see #526

**Comment by alexislefebvre** [comment]
Last 3.x release which should work with Symfony 4.3.1 (and 3.4, etc.): https://github.com/liip/LiipFunctionalTestBundle/releases/tag/3.0.0-RC2

**Comment by alexislefebvre** [comment]
Fixed by #526 and #527.
